### PR TITLE
Added feature to hide progress bar and add an instruction if there are no cards in the list #745 #746

### PR DESCRIPTION
**Pull request by Ullas-Aithal**
Added dataset oberver on adapter to hide the progress bar and add an empty view with instructions when the list is empty

### CODE REVIEW

**Comment by Ullas-Aithal** [comment]
@flocke , I've added a small instruction "Start adding your tokens by pressing '+' at the bottom" as an empty view when there are no tokens in the list. I've combined #745 and #746 since they were related. Let me know if you want me to split them or remove them or change anything else.

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -66,7 +66,19 @@
                 android:paddingTop="@dimen/activity_margin_xsmall"
                 android:paddingBottom="@dimen/fab_recyclerview_padding"
                 android:clipToPadding="false" />
-
+            <TextView
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:paddingStart="@dimen/activity_margin_small"
+                android:paddingEnd="@dimen/activity_margin_small"
```
Comment (flocke) [inline app/src/main/res/layout/activity_main.xml:73]: I would prefer `activity_margin_large` here, it just looks better in my opinion.

**Review by flocke — CHANGES_REQUESTED** [review]
I like the implementation and only have one small change request regarding the style of the empty state message.

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (Ullas-Aithal) [reply app/src/main/res/layout/activity_main.xml]: Changed it

**Review by flocke — APPROVED** [review]

**Pull request merged**